Registry.md

```diff
@@ -0,0 +1,135 @@
+# Registry Contract Documentation
+
+## Introduction
+
+The **[Registry Contract](https://github.com/horuslabsio/TBA/blob/v3/README.md#the-registry-contract)** is a key part of the Tokenbound system, enabling the deployment and management of Tokenbound Accounts (TBA) associated with NFTs. It allows for the deterministic calculation and deployment of Tokenbound Account addresses using the [ERC-6551](https://github.com/erc6551/reference) standard. This contract provides mechanisms to:
+
+- Deploy Tokenbound Accounts based on an NFT's contract address and token ID.
+- Calculate the address of a Tokenbound Account for a specific NFT without deploying it.
+- Ensure that only the NFT's owner can create an account.
+
+### Event Logging
+
+The Registry contract also logs events when a Tokenbound Account is successfully created, allowing for transparency and easy tracking of deployments.
+
+## Functions
+
+### 1. `create_account`
+
+```cairo
+fn create_account(
+    ref self: ContractState,
+    implementation_hash: felt252,
+    token_contract: ContractAddress,
+    token_id: u256,
+    salt: felt252,
+    chain_id: felt252
+) -> ContractAddress
+```
+
+### Description:
+
+This function deploys a new Tokenbound Account for an NFT. It requires that the caller be the owner of the NFT in question, ensuring that only authorized users can create Tokenbound Accounts.
+
+### Parameters (type):
+
+- `implementation_hash (felt252)`: The class hash of the reference account (implementation contract).
+- `token_contract (ContractAddress)`: The contract address of the NFT.
+- `token_id (u256)`: The unique ID of the NFT within the contract.
```

Comment:
I'd suggest the below instead of what you have right now

The unique identifier of the NFT within the token contract.